Unable to change "require" to "import" in www (or www.js)

### MDN URL

https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/Introduction

### What specific section or headline is this issue about?

Installation and first use

### What information was incorrect, unhelpful, or incomplete?

All of the examples use the old "require" convention. According to other MDN documentation, the newer "import"/"export" convention should work.

I am unable to get the top-level "www.js" file to find any module that uses export.

### What did you expect to see?

I expect "yarn start" to launch the new service without complaint.

Instead, I see something like:
```
[tms@hoyo gate-dashboard-service]$ yarn start
Debugger listening on ws://127.0.0.1:53745/66a8e394-4e26-445f-8265-da31dc1cb46e
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
yarn run v1.22.19
$ node ./bin/www.js --trace-warnings
Debugger listening on ws://127.0.0.1:53979/28084213-7c7c-4d40-a5d9-eabfe4390bb4
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
Waiting for the debugger to disconnect...
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/tms/backend/gate-dashboard-service/foobar' imported from /home/tms/backend/gate-dashboard-service/bin/www.js
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:226:11)
    at moduleResolve (node:internal/modules/esm/resolve:838:10)
    at defaultResolve (node:internal/modules/esm/resolve:1036:11)
    at DefaultModuleLoader.resolve (node:internal/modules/esm/loader:251:12)
    at DefaultModuleLoader.getModuleJob (node:internal/modules/esm/loader:140:32)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:33)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v20.5.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Waiting for the debugger to disconnect...
```

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

Here is my `package.json`:
```
{
    "name": "gate",
    "version": "0.0.0",
    "private": true,
    "type": "module",
    "scripts": {
        "start": "node ./bin/www.js --trace-warnings"
    },
    "dependencies": {
        "axios": "^1.6.7",
        "cookie-parser": "^1.4.6",
        "cors": "^2.8.5",
        "debug": "^4.3.4",
        "express": "^4.18.3",
        "express-jwt": "^8.4.1",
        "http-errors": "^2.0.0",
        "jwks-rsa": "^3.1.0",
        "morgan": "^1.10.0",
        "pug": "^3.0.2",
        "uuid": "^9.0.1"
    },
    "devDependencies": {}
}
```
Here is my stubbed version of `/bin/www.js`:
```
#!/usr/bin/env node
/**
 * Trying to replace require with import
 */

// var foobar = require('../foobar');
import { foobar } from '../foobar'
console.log(`foobar = ${JSON.stringify(foobar)}`)
```
Here is my stubbed version of `app.js` (renamed to `foobar.js`):
```
/**
 * foobar.js
 * 
 * @copyright: 2024 by Thomas M. Stambaugh & Zeetix, LLC (http://www.zeetix.com)
 * All rights reserved.
 * 
 * How does import/export work?
 */

// var foobar = 'SOME old-style export'
// module.exports = foobar

export var foobar = 'SOME new-style EXPORT';
```

Here is what I expect to see at the command line:
```
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
yarn run v1.22.19
$ node ./bin/www.js --trace-warnings
Debugger listening on ws://127.0.0.1:53275/133c3951-baf9-4dff-9d14-d98ce7048d37
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
foobar = "SOME new-style EXPORT"
Waiting for the debugger to disconnect...
Done in 0.57s.
```

Here is what I actually see:
```
[tms@hoyo gate-dashboard-service]$ yarn start
Debugger listening on ws://127.0.0.1:53932/249ee45d-a79d-4b05-8bc9-814f3047b4e6
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
yarn run v1.22.19
$ node ./bin/www.js --trace-warnings
Debugger listening on ws://127.0.0.1:53201/0a2c8bd8-9b7c-418a-9741-227a764f0a18
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
Waiting for the debugger to disconnect...
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/tms/backend/gate-dashboard-service/foobar' imported from /home/tms/backend/gate-dashboard-service/bin/www.js
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:226:11)
    at moduleResolve (node:internal/modules/esm/resolve:838:10)
    at defaultResolve (node:internal/modules/esm/resolve:1036:11)
    at DefaultModuleLoader.resolve (node:internal/modules/esm/loader:251:12)
    at DefaultModuleLoader.getModuleJob (node:internal/modules/esm/loader:140:32)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:33)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v20.5.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Waiting for the debugger to disconnect...
```
I apologize in advance if this is really really stupid or obvious, I just can't make it work. I'm migrating a mock frontend with a large number of Javascript files into a new backend service, and I'm trying to avoid having to change 285 results in 145 files to get anything to even start.


### MDN metadata

<!-- Do not make changes below this line -->
<details>
<summary>Page report details</summary>

* Folder: `en-us/learn/server-side/express_nodejs/introduction`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/Introduction
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/server-side/express_nodejs/introduction/index.md
* Last commit: https://github.com/mdn/content/commit/2528ef17bf90b1822940066c9365641181c5e8e1
* Document last modified: 2024-03-01T02:30:38.000Z

</details>

--- a/files/en-us/learn/server-side/express_nodejs/skeleton_website/index.md
+++ b/files/en-us/learn/server-side/express_nodejs/skeleton_website/index.md
@@ -366,7 +366,7 @@ The _devstart_ and _serverstart_ scripts can be used to start the same **./bin/w
 ### www file
 
 The file **/bin/www** is the application entry point! The very first thing this does is `require()` the "real" application entry point (**app.js**, in the project root) that sets up and returns the [`express()`](https://expressjs.com/en/api.html) application object.
-`require()` is a global node function that is used to import JavaScript scripts into the current file.
+`require()` is the [CommonJS way](https://nodejs.org/api/modules.html to import JavaScript code, JSON, and other files into the current file.
 Here we specify **app.js** module using a relative path and omit the optional (.**js**) file extension.
 
 ```js
@@ -379,9 +379,9 @@ Here we specify **app.js** module using a relative path and omit the optional (.
 const app = require("../app");
 ```
 
-> **Note:** Nodejs 14 and later also have stable support for the ES6 `import` statement for importing JavaScript (ECMAScript) modules.
+> **Note:** Node.js 14 and later support ES6 `import` statements for importing JavaScript (ECMAScript) modules.
 > To use this feature you have to add `"type": "module",` to your Express **package.json** file, all the modules in your application have to use `import` rather than `require()`, and for _relative imports_ you must include the file extension (for more information see the [Node documentation](https://nodejs.org/api/esm.html#introduction)).
-> While there are benefits to using `import`, this tutorial uses `require()` in order to match the official Express documentation.
+> While there are benefits to using `import`, this tutorial uses `require()` in order to match [the Express documentation](https://expressjs.com/en/starter/hello-world.html).
 
 The remainder of the code in this file sets up a node HTTP server with `app` set to a specific port (defined in an environment variable or 3000 if the variable isn't defined), and starts listening and reporting server errors and connections. For now you don't really need to know anything else about the code (everything in this file is "boilerplate"), but feel free to review it if you're interested.
 

--- a/files/en-us/learn/server-side/express_nodejs/skeleton_website/index.md
+++ b/files/en-us/learn/server-side/express_nodejs/skeleton_website/index.md
@@ -366,6 +366,8 @@ The _devstart_ and _serverstart_ scripts can be used to start the same **./bin/w
 ### www file
 
 The file **/bin/www** is the application entry point! The very first thing this does is `require()` the "real" application entry point (**app.js**, in the project root) that sets up and returns the [`express()`](https://expressjs.com/en/api.html) application object.
+`require()` is a global node function that is used to import JavaScript scripts into the current file.
+Here we specify **app.js** module using a relative path and omit the optional (.**js**) file extension.
 
 ```js
 #!/usr/bin/env node
@@ -377,7 +379,9 @@ The file **/bin/www** is the application entry point! The very first thing this 
 const app = require("../app");
 ```
 
-> **Note:** `require()` is a global node function that is used to import modules into the current file. Here we specify **app.js** module using a relative path and omitting the optional (.**js**) file extension.
+> **Note:** Nodejs 14 and later also have stable support for the ES6 `import` statement for importing JavaScript (ECMAScript) modules.
+> To use this feature you have to add `"type": "module",` to your Express **package.json** file, all the modules in your application have to use `import` rather than `require()`, and for _relative imports_ you must include the file extension (for more information see the [Node documentation](https://nodejs.org/api/esm.html#introduction)).
+> While there are benefits to using `import`, this tutorial uses `require()` in order to match the official Express documentation.
 
 The remainder of the code in this file sets up a node HTTP server with `app` set to a specific port (defined in an environment variable or 3000 if the variable isn't defined), and starts listening and reporting server errors and connections. For now you don't really need to know anything else about the code (everything in this file is "boilerplate"), but feel free to review it if you're interested.
 

--- a/files/en-us/learn/server-side/express_nodejs/skeleton_website/index.md
+++ b/files/en-us/learn/server-side/express_nodejs/skeleton_website/index.md
@@ -366,7 +366,7 @@ The _devstart_ and _serverstart_ scripts can be used to start the same **./bin/w
 ### www file
 
 The file **/bin/www** is the application entry point! The very first thing this does is `require()` the "real" application entry point (**app.js**, in the project root) that sets up and returns the [`express()`](https://expressjs.com/en/api.html) application object.
-`require()` is the [CommonJS way](https://nodejs.org/api/modules.html to import JavaScript code, JSON, and other files into the current file.
+`require()` is the [CommonJS way](https://nodejs.org/api/modules.html) to import JavaScript code, JSON, and other files into the current file.
 Here we specify **app.js** module using a relative path and omit the optional (.**js**) file extension.
 
 ```js

